File: packages/es-markdown-core/src/model/image.ts
import type { MarkupBase } from "./base.js";
import type { Markup } from "./markup.js";
import type { Destination } from "../values.js";

export interface Image extends MarkupBase<"image"> {
    /** Required, for the reason `Link.dest` is. */
    readonly dest: Destination;
    /** Optional. */
    readonly title: string | null;
    readonly width: number | null;
    readonly height: number | null;
    readonly content: readonly Markup[];
}

File: packages/kotlin-markdown-core/src/commonMain/kotlin/com/nouprax/markdown/core/model/Image.kt
package com.nouprax.markdown.core

public class Image internal constructor(
    /** Required, for the reason [Link.dest] is. */
    public val dest: Destination,
    public val title: String?,
    public val content: kotlin.collections.List<Markup>,
    override val scope: Scope,
) : Markup {
    override fun <Result> accept(visitor: Visitor<Result>): Result = visitor.visitImage(this)
}
